packages/backend/src/controllers/seasons.controller.ts

```diff
@@ -34,16 +34,25 @@ export class SeasonsController {
     return SeasonDto.fromEntity(season!);
   }
 
+  @ApiResponse({
+    status: 200,
+    description: "Returns season by id",
+    type: SeasonDto,
+  })
+  @Get("/:id")
+  async getSeasonById(@Param("id") id: string): Promise<SeasonDto> {
+    const season = await this.seasonsService.getSeason(id);
+    return SeasonDto.fromEntity(season!);
```

Comment:
instead of using `!` we should check if we got a season and throw an NotFound error otherwise

Reply:
error thrown in the service right away, see comment:
 > maybe it should be thrown in the service right away instead? then we can change the return type from the `getSeason` call to `ISeason` instead of `ISeason | null`